collector/nginx_plus.go

```diff
@@ -622,7 +623,7 @@ func (c *NginxPlusCollector) Collect(ch chan<- prometheus.Metric) {
 	c.mutex.Lock() // To protect metrics from concurrent collects
 	defer c.mutex.Unlock()
 
-	stats, err := c.nginxClient.GetStats()
+	stats, err := c.nginxClient.GetStats(context.TODO())
```

Comment:
We shouldn't use context.TODO() in production code. Can we replace with context.WithTimeout() and use a sane timeout for this API call?

Reply:
My plan was to create a separate issue to figure out if we want to give the option to pass a context or a timeout or something else. The TODO was just to make it work here (that's also why I used TODO instead of Backgorund). Does it make sense?

Reply:
Sure, that's fair. Can you add a link to the issue as a comment above this line so we don't forget?

Reply:
done